Toolbox/arcsdm/pca.py

```diff
@@ -0,0 +1,190 @@
+""" ArcSDM 6 ToolBox for ArcGIS Pro
+
+Conversion and tool development for ArcGIS Pro by Geological Survey of Finland (GTK), 2024.
+
+Compute defined number of principal components for numeric input data and transform the data.
+
+Before computation, data is scaled according to specified scaler and NaN values removed or replaced.
+Optionally, a nodata value can be given to handle similarly as NaN values.
+
+This tool is based on the PCA implementation in the scikit-learn library originally developed by University of Turku.
```

Comment:
Nice 👍  A minor suggestion - add at the end that the original implementation is included in the package eis_toolkit?